343 standardize on a single icon library to reduce bundle size

yvanhenang opened the pull request.

yvanhenang committed:
Complete icon migration to react-feather

yvanhenang committed:
chore(ci): update workflow to run on develop branch instead of main

yvanhenang committed:
Migrate icons to react-feather and fix build issues

yvanhenang committed:
Apply Prettier auto-formatting to all files

yvanhenang committed:
fix: resolve stale react-icons/fa import error by cleaning cache and reinstalling deps

yvanhenang committed:
style: format code with Prettier for DocumentImages and RecoverAccountPage

yvanhenang committed:
refactor: use optional chaining for phone number validation

yvanhenang committed:
style: format DocumentImages and SideBar components with Prettier

yvanhenang committed:
refactor: use a more stable key for OTP input fields

yvanhenang committed:
style: format SideBar component with Prettier

yvanhenang committed:
refactor: use Tailwind for styles in ContactList, Header, and Installbutton

yvanhenang committed:
chore: format code with Prettier

yvanhenang committed:
feat: migrate all icons to react-feather and clean up old icon libraries

yvanhenang committed:
Migrate all icons to react-feather and remove old icon libraries.

yvanhenang committed:
docs: update icon library comment to reflect react-feather usage

NkwaTambe commented on Docs/Github_workflow_overview.md:
```diff
@@ -92,7 +92,7 @@ The workflow is set to run on:
 - **Build Failures**: Ensure code changes do not introduce syntax errors or unsupported constructs. Check for environment-specific build configurations.
-- **Lint/Prettier Errors**: Ensure that the code follows the project's style guide and run `npm run prettier:check -- --write "src/**/*.{js,jsx,ts,tsx,json,css,scss,md}"` locally to auto-correct issues.
+- **Lint/Prettier Errors**: Ensure that the code follows the project's style guide and run  locally to auto-correct issues.
 - **TypeScript Errors**: Resolve any type errors reported in the `typescript` step by fixing the type definitions in the code.
```
This change is not neccessary.. please update to original state

yvanhenang committed:
refactor: migrate all icons to react-feather

yvanhenang committed:
Fix:put back the changes I wasn't supposed to touch

NkwaTambe commented on src/shared/components/SideBar.tsx:
```diff
@@ -1,173 +1,87 @@
-    setAcknowledgedAdmin(true);
-  };
-
   return (
```
This are important parts of our application, please look into it... focus on changing the icons and not application logic

NkwaTambe replied:
This are important parts of our application, please look into it... focus on changing the icons and not application logic

NkwaTambe commented on src/shared/components/SideBar.tsx:
```diff
@@ -1,173 +1,87 @@
-                  </p>
-                  <div className="mt-4">
-                    <button
-                      onClick={handleContinue}
-                      className="bg-blue-500 text-white px-4 py-2 rounded hover:bg-blue-600 mr-2"
-                    >
-                      I am an administrator
-                    </button>
-                    <button
-                      onClick={onClose}
-                      className="bg-gray-300 text-gray-700 px-4 py-2 rounded hover:bg-gray-400"
-                    >
-                      Close
-                    </button>
-                  </div>
-                </div>
-              </div>
-            </div>
-          </div>
-        ) : (
-          <div className="p-4">
-            {menuItems.map((item) => (
-              <button
-                key={item.id}
-                onClick={() => {
-                  item.onClick();
-                  onClose();
-                }}
-                className="flex items-center w-full p-4 text-left hover:bg-gray-100 rounded-lg transition-colors mb-2"
-              >
-                <FontAwesomeIcon
-                  icon={item.icon}
-                  className="text-blue-500 mr-3"
-                />
-                <span>{item.label}</span>
-              </button>
-            ))}
-          </div>
-        )}
-        <div className="h-16"></div>
```
Administrative control is needed ... please put it back or make it better

hanswizzy commented on src/features/kyc/pages/KycRecoveryPage.tsx:
```diff
@@ -168,7 +168,11 @@ export default function RecoveryDashboard() {
       >
-        <FiArrowLeft className="w-6 h-6 text-gray-600" />
+        <ArrowLeft
+          size={24}
+          color="#6B7280"
+          style={{ width: "1.5rem", height: "1.5rem" }}
```
use tailwind css , why change to custom styling?

hanswizzy commented on src/features/kyc/pages/KycRecoveryPage.tsx:
```diff
@@ -213,7 +217,11 @@ export default function RecoveryDashboard() {
               >
-                <FiArrowLeft className="w-5 h-5" />
+                <ArrowLeft
+                  size={24}
+                  color="#6B7280"
+                  style={{ width: "1.5rem", height: "1.5rem" }}
```
same here!

hanswizzy commented on src/features/kyc/guidelines/GuidelinesPage.tsx:
```diff
@@ -193,14 +195,21 @@ const GuidelinesPage: React.FC = () => {
-            <FaCheckCircle
-              className="mr-3 text-blue-600 flex-shrink-0"
+            <Info
+              size={20}
+              color="#2563EB"
+              style={{ marginRight: 12 }}
```
same here, use tailwind css instead!

hanswizzy commented on src/features/kyc/guidelines/GuidelinesPage.tsx:
```diff
@@ -50,26 +44,34 @@ const MarkdownH2: React.FC<MarkdownHeadingProps> = ({ children, ...props }) => {
       {typeof firstChild === "string" && firstChild.startsWith("Step 1") && (
-        <FaCamera
-          className="mr-3 text-blue-600 text-xl flex-shrink-0"
+        <Info
+          size={20}
+          color="#2563EB"
+          style={{ marginRight: 12 }}
```
same use tailwind css

hanswizzy commented on src/features/kyc/guidelines/GuidelinesPage.tsx:
```diff
@@ -50,26 +44,34 @@ const MarkdownH2: React.FC<MarkdownHeadingProps> = ({ children, ...props }) => {
       {typeof firstChild === "string" && firstChild.startsWith("Step 2") && (
-        <FaShareAlt
-          className="mr-3 text-blue-600 text-xl flex-shrink-0"
+        <Info
+          size={20}
+          color="#2563EB"
+          style={{ marginRight: 12 }}
```
same

hanswizzy commented on src/features/kyc/guidelines/GuidelinesPage.tsx:
```diff
@@ -50,26 +44,34 @@ const MarkdownH2: React.FC<MarkdownHeadingProps> = ({ children, ...props }) => {
       {typeof firstChild === "string" && firstChild.startsWith("Step 3") && (
-        <FaCheckCircle
-          className="mr-3 text-blue-600 text-xl flex-shrink-0"
+        <Info
+          size={20}
+          color="#2563EB"
+          style={{ marginRight: 12 }}
```
same

hanswizzy commented on src/features/kyc/guidelines/GuidelinesPage.tsx:
```diff
@@ -50,26 +44,34 @@ const MarkdownH2: React.FC<MarkdownHeadingProps> = ({ children, ...props }) => {
-        <FaLightbulb
-          className="mr-3 text-amber-500 text-xl flex-shrink-0"
+        <Info
+          size={20}
+          color="#F59E42"
+          style={{ marginRight: 12 }}
```
same

hanswizzy commented on src/shared/components/share-handler/ErrorState.tsx:
```diff
@@ -1,69 +1,33 @@
-          </button>
-        </div>
-
-        <div className="mt-6 p-4 bg-gray-100 rounded-lg text-left">
-          <h3 className="font-bold mb-2">Troubleshooting Steps:</h3>
-          <ul className="list-disc pl-6 space-y-2">
-            <li>Ensure you're not in private/incognito mode.</li>
-            <li>Enable third-party cookies in browser settings.</li>
-            <li>Click "Load Test Data" to simulate shared content.</li>
-            <li>Share a file from your device to this app.</li>
-            <li>Ensure the app is running over HTTPS.</li>
-            <li>Contact support if the issue persists.</li>
-          </ul>
-        </div>
-      </div>
```
why remove this?

NkwaTambe commented on src/shared/components/SideBar.tsx:
```diff
@@ -1,173 +1,87 @@
-            tellerAccountCert: accountCert,
-            redirectTo: "/agent-topup",
-          },
-        }),
```
This are important parts of our application, please look into it... focus on changing the icons and not application logic

yvanhenang committed:
Replace all FontAwesome icons with react-feather icons

yvanhenang committed:
Fix code formatting with Prettier

stephane-segning commented on Docs/Arc42.md:
Cancel these changes

stephane-segning commented on Docs/Github_workflow_overview.md:
Cancel these changes

stephane-segning commented on Docs/Production/GrafanaMonitoring.md:
Cancel these changes

stephane-segning commented on Docs/Sonarqube-Integration.md:
Cancel these changes

stephane-segning commented on public/sw.js:
Delete this file

stephane-segning reviewed (changes requested):
You should have the same formatting as the others, else you'll have free conflicts for no reason. Please fix the formatting.

yvanhenang committed:
Fix the review

yvanhenang committed:
Resolve conflict

yvanhenang committed:
Resolve conflict

yvanhenang committed:
Delete sw.js file

hanswizzy committed:
Merge branch 'develop' into 343-standardize-on-a-single-icon-library-to-reduce-bundle-size

Koufan-De-King committed:
fixed conflicts with routing branch

Koufan-De-King committed:
Merge remote-tracking branch 'origin/develop' into 343-standardize-on-a-single-icon-library-to-reduce-bundle-size

Koufan-De-King committed:
fixed settings page text alignment

Koufan-De-King committed:
removed laszy loading, removed manual chunking, and added sw caching in order to enhance offline functionality

yvanhenang committed:
Improve KYC form layout and responsiveness on RecoverAccountPage

- Increased form container width for better content display in FormComponents.tsx
- Adjusted background layout and spacing in RecoverAccountPage.tsx

yvanhenang committed:
Fix prettier:check

Koufan-De-King committed:
Merge branch 'enhanced-caching-to-reinstate-offline-functionality' into 343-standardize-on-a-single-icon-library-to-reduce-bundle-size

Koufan-De-King committed:
amended recovery process

Koufan-De-King committed:
fixed offline withdrawal

Koufan-De-King committed:
fixed minor qr and toast issues

Koufan-De-King committed:
fixed minor qr and toast issues

Koufan-De-King committed:
fixed sonar issues
